legacy/ui/legacy/src/main/java/com/fsck/k9/activity/MessageList.kt

```diff
@@ -407,10 +407,14 @@ open class MessageList :
             }
         }
     }
-
     private fun decodeExtras(intent: Intent): Boolean {
         val launchData = decodeExtrasToLaunchData(intent)
-
+        if (intent.action === Intent.ACTION_SEARCH && !intent.component?.className.equals(
+                "com.fsck.k9.activity.Search",
+            )
+        ) {
```

Comment:
I would like to suggest the following change:
```suggestion
        if (intent.action == Intent.ACTION_SEARCH && intent.component?.className != Search::class.java.name) {
```

It will be easier to track whether the name or package of the `Search` Activity changes in the future. Additionally, as we are finishing the Activity if the intent does not originate from the `Search` Activity, I recommend moving the conditional check before the `decodeExtrasToLaunchData(intent)`, as we won’t need that data at that point.